lib/ecto/type.ex

```diff
@@ -301,92 +296,124 @@ defmodule Ecto.Type do
       iex> dump({:array, :binary}, ["1", "2", "3"])
       {:ok, ["1", "2", "3"]}
 
-  A `dumper` function may be given for handling recursive types.
+  A `dumper` function may be given for handling recursive types, see: `dump/3`.
```

Comment:
`dump/2` does not include the `dumper` argument, so I would replace the ``` `dumper` ``` reference in this paragraph with something like “An additional function” or similar :)